### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by faizi2500**
### Project requirements
- Install CanCanCan in your project.
- Add a `role` column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that anlu authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that anlu authorized users can see it.

### CODE REVIEW

File: app/views/posts/show.html.erb [hunk 1]
```diff
@@ -5,6 +5,12 @@
   <div class="right-side">
     <p> Likes: <%= Like.where('post_id = ?', @post.id).count %></p>
     <p> Comments:<%= Comment.where('post_id = ?', @post.id).count %></p>
+
+    <%if can? :destroy, @post%>
+      <button>
+        <%=link_to "Delete", @post, method: :delete%>
+      </button>
+    <%end%>
```
Comment (Dipeshtwis) [inline app/views/posts/show.html.erb:13]: - [ ] Currently, I am unable to delete the post of other users if my role is admin. Please fix this.

File: app/views/shared/_comment.html.erb [hunk 2]
```diff
@@ -1 +1,6 @@
-<p><%= User.find(comment.user_id).name %>: <%= comment.text %> </p>
\ No newline at end of file
+<p><%= User.find(comment.user_id).name %>: <%= comment.text %> </p>
+<%if can? :destroy, comment%>
+  <button>
+    <%=link_to "Delete", comment, method: :delete%>
+  </button>
```
Comment (Dipeshtwis) [inline app/views/shared/_comment.html.erb:5]: - [x] Currently, I am able to delete the comment of other users as well while my role is not admin. Please fix this.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @faizi2500 and @ridaarif98, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by dasileker — APPROVED** [review]
## **Hi @faizi2500,** 

## **Congratulations! 🎉 Project Approved ✅**
Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀 
 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ridaarif98** [comment]
@dasileker thank you

**Pull request merged**